db/version_set.cc

```diff
@@ -4708,6 +4709,92 @@ Status VersionSet::ReduceNumberOfLevels(const std::string& dbname,
       mutable_cf_options, &ve, &dummy_mutex, nullptr, true);
 }
 
+// Get the checksum information including the checksum and checksum function
+// name of all SST files in this Manifest. Store the information in
+// FileChecksumList which contains a map from file number to its checksum info.
+Status VersionSet::GetAllFileChecksumInfo(const Options& options,
```

Comment:
Gave it another thought. Do you intend to call `DB::GetAllFileChecksumsInfo()` on a live DB? If so, the file metadata info should have already be in memory, thus you don't have to read the MANIFEST.
If you also use this for ldb too, you can always call `Recover()` first.
My suggestion is, maybe you can remove the MANIFEST reading part from this method. Instead, can you populate the checksum info from column families' version storage info?